### PR TITLE
merge metadata based on the type of the values

### LINKED ISSUE #20: Multiple meta files behavior
Hi, I would like to know which is the default behavior when multiple `.meta` files are used when calling the build. For example if I call `colcon build --metas conf1.meta conf2.meta `  with the following files:

conf1.meta
```
{
    "names": {
        "package1": {
            "cmake-args": [
                "-DOPTION_1=1",
                "-DOPTION_2=2",
                "-DOPTION_3=3",
            ]
        }
    }
}
```

conf2.meta
```
{
    "names": {
        "package1": {
            "cmake-args": [
                "-DOPTION_2=22",
                "-DOPTION_4=4",
            ]
        }
    }
}
```

Are all the options being merged and duplicated ones being overwritted with order-based priority?

Thanks

### PR DESCRIPTION
**Pull request by dirk-thomas**
The more sophisticated merge resolves #20.

### CODE REVIEW

File: colcon_metadata/metadata/__init__.py [hunk 1]
```diff
@@ -46,3 +49,67 @@ def get_metadata_files(path=None):
             path = os.path.join(dirpath, filename)
             files.append(path)
     return files
+
+
+def merge_metadata(dst_dict, other_dict, meta_source):
```
Comment (jacobperron) [inline colcon_metadata/metadata/__init__.py]: `other_dict` -> `src_dict` might make it more clear.

File: colcon_metadata/metadata/__init__.py [hunk 2]
```diff
@@ -46,3 +49,67 @@ def get_metadata_files(path=None):
...
+    :param str meta_source: The source of the metadata, only used for log
+      messages
+    """
+    for key, value in other_dict.items():
+        merge_metadata_value(dst_dict, key, value, meta_source)
+
+
+def merge_metadata_value(dict_, key, value, meta_source):
+    """
+    Set or update an key of a dictionary.
```
Comment (jacobperron) [inline colcon_metadata/metadata/__init__.py]:
```suggestion
    Set or update a key of a dictionary.
```

**Review by jacobperron — COMMENTED** [review]
Two minor comments, otherwise LGTM

File: colcon_metadata/metadata/__init__.py (hunk 1 above)
Comment (dirk-thomas) [reply colcon_metadata/metadata/__init__.py]: Renamed: 4aeba30f2d58aee2f9f92c82b25858ac0b4db235.

**Review by jacobperron — APPROVED** [review]

**Pull request merged**